src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -64,69 +64,92 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
-                            // hide NavViewItem pointing to "Add Profile"
-                            navViewItem.Visibility(Visibility::Collapsed);
+                            // don't add the "Add Profile" item
+                            continue;
                         }
                     }
                 }
             }
+            menuItemsSTL.emplace_back(item);
         }
-        _InitializeProfilesList();
+        menuItems.ReplaceAll(menuItemsSTL);
 
-        _RefreshCurrentPage();
-    }
+        // Repopulate profile-related menu items
+        _InitializeProfilesList();
 
-    void MainPage::_RefreshCurrentPage()
-    {
-        auto navigationMenu{ SettingsNav() };
-        if (const auto selectedItem{ navigationMenu.SelectedItem() })
+        // now that the menuItems are repopulated,
+        // refresh the current page using the SelectedItem data we collected before the refresh
+        if (selectedItemTag)
         {
-            if (const auto tag{ selectedItem.as<MUX::Controls::NavigationViewItem>().Tag() })
+            const auto& selectedItemStringTag{ selectedItemTag.try_as<hstring>() };
+            const auto& selectedItemProfileTag{ selectedItemTag.try_as<ProfileViewModel>() };
+            for (const auto& item : menuItems)
             {
-                if (const auto oldProfile{ tag.try_as<Editor::ProfileViewModel>() })
+                if (const auto& menuItem{ item.try_as<MUX::Controls::NavigationViewItem>() })
                 {
-                    // check if the profile still exists
-                    if (const auto profile{ _settingsClone.FindProfile(oldProfile.Guid()) })
+                    if (const auto& tag{ menuItem.Tag() })
                     {
-                        // Navigate to the page with the given profile
-                        _Navigate(_viewModelForProfile(profile));
-                        return;
+                        if (const auto& stringTag{ tag.try_as<hstring>() })
+                        {
+                            if (stringTag == selectedItemStringTag)
+                            {
+                                // found the one that was selected before the refresh
+                                SettingsNav().SelectedItem(item);
+                                _Navigate(*stringTag);
+                                co_return;
+                            }
+                        }
+                        else if (const auto& profileTag{ tag.try_as<ProfileViewModel>() })
+                        {
+                            if (profileTag->Guid() == selectedItemProfileTag->Guid())
+                            {
+                                // found the one that was selected before the refresh
+                                SettingsNav().SelectedItem(item);
+                                _Navigate(*selectedItemProfileTag);
+                                co_return;
+                            }
+                        }

```

Comment:
hey how will this work when you have BASE LAYER selected?

Reply:
Base Layer is unique in that it has a string tag. So that's covered in the other if-statement.